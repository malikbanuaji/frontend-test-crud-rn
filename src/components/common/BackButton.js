import React from 'react';
import {StyleSheet, TouchableNativeFeedback, View} from 'react-native';
import Icon from 'react-native-vector-icons/Feather';
import {Mixins} from '../../styles';

export default function BackButton({onPress}) {
  return (
    <View style={styles.backButton}>
      <TouchableNativeFeedback onPress={onPress}>
        <View style={styles.button}>
          <Icon name={'chevron-left'} size={24} />
        </View>
      </TouchableNativeFeedback>
    </View>
  );
}

const styles = StyleSheet.create({
  backButton: {
    ...Mixins.baseRounded,
  },
  button: {
    ...Mixins.backButtonDimension,
    justifyContent: 'center',
    alignItems: 'center',
  },
});
