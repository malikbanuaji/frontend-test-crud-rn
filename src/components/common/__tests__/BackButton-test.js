import React from 'react';

import {fireEvent, render} from '@testing-library/react-native';
import BackButton from '../BackButton';

test('renders correctly', () => {
  const tree = render(<BackButton />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('handle press correctly', () => {
  const mockOnPress = jest.fn();
  const {getByTestId} = render(<BackButton onPress={mockOnPress} />);
  fireEvent.press(getByTestId('back-button'));
  expect(mockOnPress).toBeCalled();
});
