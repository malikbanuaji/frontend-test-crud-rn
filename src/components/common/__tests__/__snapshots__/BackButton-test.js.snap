// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "borderRadius": 8,
      "overflow": "hidden",
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "height": 30,
        "justifyContent": "center",
        "width": 30,
      }
    }
    testID="back-button"
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 24,
          },
          undefined,
          Object {
            "fontFamily": "Feather",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      ï„¯
    </Text>
  </View>
</View>
`;
